playground/CustomResources/CustomResources.AppHost/TestResource.cs

```diff
@@ -28,53 +27,44 @@ public static IResourceBuilder<TestResource> AddTestResource(this IDistributedAp
     }
 }
 
-internal sealed class TestResourceLifecycleHook : IDistributedApplicationLifecycleHook, IAsyncDisposable
+internal sealed class TestResourceLifecycleHook(ResourceNotificationService notificationService, ResourceLoggerService loggerService) : IDistributedApplicationLifecycleHook, IAsyncDisposable
```

Comment:
I see that `AddTestResource` calls `builder.Services.AddLifecycleHook<TestResourceLifecycleHook>()`. What happens each time that call is made? Will it register multiple? Should it call TryAdd instead?

FYI `AddAzureProvisioning` calls `AddLifecycleHook`. Should it call TryAdd instead?

Reply:
Should be try adds.

Reply:
Fixed